### ISSUE TITLE
Segmentation fault during program start - Ubuntu 20.04

### ISSUE THREAD

**Issue by lucasmr** [obsproject/obs-studio#3755]
## Platform

Operating system and version: Ubuntu 20.04
OBS Studio version: 26.0.2

## Expected Behavior

OBS should start

## Current Behavior

OBS is not starting. Tried purging the install and removing ~/.config/obs-studio folders, but a fresh install is not fixing it.

## Steps to Reproduce

1. Run OBS.

## Additional information

```
$ glxinfo | grep 'OpenGL\|direct '
direct rendering: Yes
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) HD Graphics 620 (KBL GT2)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 20.0.8
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 4.6 (Compatibility Profile) Mesa 20.0.8
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.2 Mesa 20.0.8
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
OpenGL ES profile extensions:

$ cat /var/log/Xorg.0.log | grep modesetting
[     5.757] (II) LoadModule: "modesetting"
[     5.758] (II) Loading /usr/lib/xorg/modules/drivers/modesetting_drv.so
[     5.759] (II) Module modesetting: vendor="X.Org Foundation"
[     5.759] (II) modesetting: Driver for Modesetting Kernel Drivers: kms

$ obs
Attempted path: share/obs/obs-studio/locale/en-US.ini
Attempted path: /usr/share/obs/obs-studio/locale/en-US.ini
Attempted path: share/obs/obs-studio/locale.ini
Attempted path: /usr/share/obs/obs-studio/locale.ini
Attempted path: share/obs/obs-studio/themes/Dark.qss
Attempted path: /usr/share/obs/obs-studio/themes/Dark.qss
info: CPU Name: Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
info: CPU Speed: 1803.372MHz
info: Physical Cores: 2, Logical Cores: 4
info: Physical Memory: 15900MB Total, 9291MB Free
info: Kernel Version: Linux 5.4.0-54-generic
info: Distribution: "Ubuntu" "20.04"
info: Window System: X11.0, Vendor: The X.Org Foundation, Version: 1.20.8
info: Portable mode: false
Attempted path: share/obs/obs-studio/themes/Dark/no_sources.svg
Attempted path: /usr/share/obs/obs-studio/themes/Dark/no_sources.svg
QMetaObject::connectSlotsByName: No matching signal for on_advAudioProps_clicked()
QMetaObject::connectSlotsByName: No matching signal for on_advAudioProps_destroyed()
QMetaObject::connectSlotsByName: No matching signal for on_actionGridMode_triggered()
QMetaObject::connectSlotsByName: No matching signal for on_program_customContextMenuRequested(QPoint)
QMetaObject::connectSlotsByName: No matching signal for on_transitionRemove_clicked()
info: OBS 26.0.2 (linux)
info: ---------------------------------
info: ---------------------------------
info: audio settings reset:
	samples per sec: 48000
	speakers:        2
info: ---------------------------------
info: Initializing OpenGL...
Segmentation fault (core dumped)

$ dmesg | tail
[ 1148.693957] obs[5251]: segfault at 7f87d042ad30 ip 00007f87d042ad30 sp 00007ffcaded6338 error 14 in libobsglad.so.0[7f87d04e3000+3f000]
[ 1148.693964] Code: Bad RIP value.
```

**Comment by Fenrirthviti** [comment]
Please provide a full gdb backtrace for the crash as well.

**Comment by lucasmr** [comment]
Building the source locally fixes it. I'm afraid there is no -dbg package in the repository.

**Comment by kkartaltepe** [comment]
I think you are confused, there is no -dbg packaging has nothing to do with providing a backtrace.

**Comment by lucasmr** [comment]
Weird. I uninstalled everything again, and after reinstalling the package, now it runs.

I presume some config file was the cause, because the compiled version fixed. I will close the issue because I can't reproduce anymore.

**Comment by MiroJanosik** [comment]
I have encountered the same problem; OBS was running at first, then after doing some package upgrades it stopped.
I have run 'gdb obs', 'run' and then 'bt' whit this output, does it help:

:~$ gdb obs
GNU gdb (Ubuntu 9.2-0ubuntu1~20.04) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from obs...
(No debugging symbols found in obs)
(gdb) run
Starting program: /usr/bin/obs 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffeb313700 (LWP 24034)]
[New Thread 0x7fffe9e54700 (LWP 24035)]
[New Thread 0x7fffe9653700 (LWP 24036)]
[New Thread 0x7fffe8e52700 (LWP 24037)]
[New Thread 0x7fffdbfff700 (LWP 24038)]
Attempted path: share/obs/obs-studio/locale/en-US.ini
Attempted path: /usr/share/obs/obs-studio/locale/en-US.ini
Attempted path: share/obs/obs-studio/locale.ini
Attempted path: /usr/share/obs/obs-studio/locale.ini
Attempted path: share/obs/obs-studio/locale/sk-SK.ini
Attempted path: /usr/share/obs/obs-studio/locale/sk-SK.ini
info: Using preferred locale 'sk-SK'
Attempted path: share/obs/obs-studio/themes/Dark.qss
Attempted path: /usr/share/obs/obs-studio/themes/Dark.qss
info: CPU Name: Intel(R) Core(TM) i7 CPU         920  @ 2.67GHz
info: CPU Speed: 2232.564MHz
info: Physical Cores: 4, Logical Cores: 8
info: Physical Memory: 11977MB Total, 7168MB Free
info: Kernel Version: Linux 5.4.0-71-generic
info: Distribution: "Ubuntu" "20.04"
info: Session Type: x11
info: Window System: X11.0, Vendor: The X.Org Foundation, Version: 1.20.9
[New Thread 0x7fffdb7fe700 (LWP 24039)]
info: Portable mode: false
[New Thread 0x7fffdaffd700 (LWP 24040)]
Attempted path: share/obs/obs-studio/themes/Dark/no_sources.svg
Attempted path: /usr/share/obs/obs-studio/themes/Dark/no_sources.svg
QMetaObject::connectSlotsByName: No matching signal for on_advAudioProps_clicked()
QMetaObject::connectSlotsByName: No matching signal for on_advAudioProps_destroyed()
QMetaObject::connectSlotsByName: No matching signal for on_actionGridMode_triggered()
QMetaObject::connectSlotsByName: No matching signal for on_program_customContextMenuRequested(QPoint)
QMetaObject::connectSlotsByName: No matching signal for on_transitionRemove_clicked()
info: OBS 26.1.1 (linux)
info: ---------------------------------
info: ---------------------------------
info: audio settings reset:
	samples per sec: 48000
	speakers:        2
[New Thread 0x7fffda7fc700 (LWP 24041)]
info: ---------------------------------
info: Initializing OpenGL...
[New Thread 0x7fffc7fff700 (LWP 24042)]
[New Thread 0x7fffc77fe700 (LWP 24043)]
[New Thread 0x7fffc6ffd700 (LWP 24044)]
[New Thread 0x7fffc67fc700 (LWP 24045)]
[New Thread 0x7fffc5ffb700 (LWP 24046)]
[New Thread 0x7fffc57fa700 (LWP 24047)]
[New Thread 0x7fffc4ff9700 (LWP 24048)]
[New Thread 0x7fffbbdb0700 (LWP 24049)]
[New Thread 0x7fffbb5af700 (LWP 24050)]
[New Thread 0x7fffbadae700 (LWP 24051)]
[New Thread 0x7fffba5ad700 (LWP 24052)]
[New Thread 0x7fffb9dac700 (LWP 24053)]

Thread 1 "obs" received signal SIGSEGV, Segmentation fault.
0x00007fffe803dd30 in ?? ()
(gdb) bt
#0  0x00007fffe803dd30 in  ()
#1  0x00007fffe816dc8e in gl_platform_create () at /usr/lib/libobs-opengl.so.0
#2  0x00007fffe81779b9 in device_create () at /usr/lib/libobs-opengl.so.0
#3  0x00007ffff5dc2c63 in gs_create () at /usr/lib/libobs.so.0
#4  0x00007ffff5e0cefd in obs_reset_video () at /usr/lib/libobs.so.0
#5  0x0000555555614d4f in  ()
#6  0x0000555555630099 in  ()
#7  0x00005555555fd2c7 in  ()
#8  0x00005555555d56c0 in main ()
(gdb)

**Comment by MiroJanosik** [comment]
Interesting. So, I wanted to build the OBS myself; but I had to upgrade from Ubuntu 20.04 to 20.10 (because of some dependency was not available as 0.3.x but only 0.2.x).
After update the OBS still does not run.
Then, I downloaded all the build essentials, dependencies, started the build.

And now, the OBS installed from the repository (not the built one!) works! No more segfault. It works the same way, also OBS gets stuck when starting virtual camera as it was before.

Interesting.
